src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -50,6 +50,12 @@ public static string HtmlEncode(string value)
             return StringBuilderCache.GetStringAndRelease(sb);
         }
 
+        public static void HtmlEncode(string value, TextWriter output)
+        {
+            // Since we already have a private method that decodes the value using a StringBuilder we don't want to have duplicate code in order to have a better mantainance.
+            output.Write(HtmlEncode(value));
```

Comment:
This would be better than the approach we discussed earlier, right? @AlexGhiondea 

Just to give you a little bit of context @justinvp, we were thinking of creating an internal abstract class `WriterBaseClass` (this is just a random name), and making two other internal classes that would inherit from this one, those to classes would be:

`SBWrapper`and `TextWriterWrapper`where them would override virtual `write(char)`and `write(string)` methods from `WriterBaseClass`that would write either to the `StringBuilder`or the `TextWriter`.

`SBWrapper` would override `ToString` and call `StringBuilderCache.GetStringAndRelease(sb)`.

But I think your approach works better cause this one would have extra allocations as well and would have to make virtual calls.